aspnetcore/mvc/models/validation.md

```diff
@@ -1,79 +1,139 @@
+Non-nullable types and strings are handled differently on the client compared to the server. On the client:
 
-MVC model binding, which isn't concerned with validation and validation attributes, rejects a form field submission containing a missing value or whitespace for a non-nullable type. In the absence of a `BindRequired` attribute on the target property, model binding ignores missing data for non-nullable types, where the form field is absent from the incoming form data.
+* A value is considered present only if input is entered for it. Therefore, client-side validation handles non-nullable types the same as nullable types.
+* Whitespace in a string field is considered valid input by the jQuery Validation [required](https://jqueryvalidation.org/required-method/) method. Server-side validation considers a required string field invalid if only whitespace is entered.
 
-The [BindRequired attribute](/dotnet/api/microsoft.aspnetcore.mvc.modelbinding.bindrequiredattribute) (also see <xref:mvc/models/model-binding#customize-model-binding-behavior-with-attributes>) is useful to ensure form data is complete. When applied to a property, the model binding system requires a value for that property. When applied to a type, the model binding system requires values for all of the properties of that type.
+As noted earlier, non-nullable types are treated as though they had a `[Required]` attribute. That means you get client-side validation even if you don't apply the `[Required]` attribute. But if you don't use the attribute, you get a default error message. To specify a custom error message, use the attribute.
 
-When you use a [Nullable\<T> type](/dotnet/csharp/programming-guide/nullable-types/) (for example, `decimal?` or `System.Nullable<decimal>`) and mark it `Required`, a server-side validation check is performed as if the property were a standard nullable type (for example, a `string`).
+### [BindRequired] attribute
 
-Client-side validation requires a value for a form field that corresponds to a model property that you've marked `Required` and for a non-nullable type property that you haven't marked `Required`. `Required` can be used to control the client-side validation error message.
+Server-side validation is done after model binding. To catch missing input for a non-nullable type during model binding, use the [BindRequired](xref:Microsoft.AspNetCore.Mvc.ModelBinding.BindRequiredAttribute) attribute. This attribute requires that data be submitted for model binding. When applied to a type, values for all of that type's properties are required. For more information, see <xref:mvc/models/model-binding#customize-model-binding-behavior-with-attributes>.
 
 ::: moniker range=">= aspnetcore-2.1"
 
+## Custom attributes
```

Comment:
Why aren't this and the next section subsections of `### Alternatives to built-in attributes` or at least immediately after?